core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -181,6 +185,34 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       else consumer.pause(java.util.Collections.singleton(tp))
     }
 
+    def tryPoll(timeout: Long, onSuccess: ConsumerRecords[K, V] => Unit): Unit = {
+      Try(consumer.poll(timeout))
+        .transform(
+          rawResult => {
+            wakeupTask.cancel()
+            wakeups = 0
+            Success(onSuccess(rawResult))
+          },
+          f => {
+            wakeupTask.cancel()
+            Failure(f)
+          }
+        )
+        .recover {
+          case w: WakeupException =>
+            wakeups = wakeups + 1
```

Comment:
I don't think we need to wait before calling another `poll()`. If the second poll hangs, we have a timeout which will define for how long until next `WakeupException`.